docs/components/rendering_iterables.md

```diff
@@ -1,46 +1,106 @@
 ```python exec
 import reflex as rx
 
-from pcweb.pages.docs import vars
+from pcweb.pages import docs
 ```
 
 # Rendering Iterables
 
-You will often want to display multiple similar components from a collection of data. The `rx.foreach` component takes an `iterable` (list, tuple or dict) and a `function` that renders each item in the list. This is useful for dynamically rendering a list of items defined in a state.
-
-In this first simple example we iterate through a `list` of colors and render the name of the color and use this color as the background for that `rx.box`. As we can see we have a function `colored_box` that we pass to the `rx.foreach` component. This function renders each item from the `list` that we have defined as a state var `color`.
+Recall again from the [basics]({docs.getting_started.basics.path}) that we cannot use Python `for` loops when referencing state vars in Reflex. Instead, use the `rx.foreach` component to render components from a collection of data.
 
 ```python demo exec
 class IterState(rx.State):
     color: list[str] = [
         "red",
         "green",
         "blue",
-        "yellow",
-        "orange",
-        "purple",
     ]
 
 
 def colored_box(color: str):
```

Comment:
how come this example has the typing as `str` but later under the render function section we specifically state that "type annotation for the color parameter in the colored_box function is rx.Var[str] (rather than just str). "